src/cli/init.rs

```diff
@@ -49,14 +53,25 @@ fn generate_config() -> Result<AppChainConfig, InitError> {
     let base_path = get_text_input("Enter base path for data directory of your app chain:", Some(default_base_path))?;
     let chain_id = get_text_input("Enter chain id for your app chain:", Some("MADARA"))?;
     let mode = get_option("Select mode for your app chain:", RollupMode::iter().collect::<Vec<_>>())?;
-    let da_layer = get_option("Select DA layer for your app chain:", DALayer::iter().collect::<Vec<_>>())?;
+    let da_layer_type = get_option("Select DA layer for your app chain:", DALayer::iter().collect::<Vec<_>>())?;
     let block_time =
         get_custom_input::<u64>("Enter block time of chain:", Some(1000), Some("Time in ms (e.g, 1000, 2000)."))?;
     let disable_fees = get_boolean_input("Do you want to disable fees for your app chain:", Some(false))?;
     let fee_token = get_text_input("Enter fee token:", Some("STRK"))?;
     let madara_version = get_latest_commit_hash(MADARA_REPO_ORG, MADARA_REPO_NAME)?;
     let config_version = ConfigVersion::Version1;
 
+    let result = match da_layer_type {
```

Comment:
So over here, what we should ideally do is create multiple structs

- AvailConfig
- CelestiaConfig
- EthereumDaConfig

Then we should have a trait called `DaConfig` which all these structs should implement. It will have functions like `generate_key_pair`, `generate_config` etc.

So in init.rs, we can directly create an struct that implements `DaConfig`. This struct could be anything, avail, celestia etc. and will only be known at compile time. But since we know it implements the trait `DaConfig`, we can call methods like `generate_key_pair`.

This is basically a factory pattern, you can read this on how its done in rust - https://refactoring.guru/design-patterns/factory-method/rust/example